src/Aspire.Hosting.Azure/api/Aspire.Hosting.Azure.cs

```diff
@@ -205,6 +220,26 @@ public BicepTemplateFile(string path, bool deleteFileOnDispose) { }
         public readonly void Dispose() { }
     }
 
+    public sealed partial class ExistingAzureResourceAnnotation : ApplicationModel.IResourceAnnotation
+    {
+        public ExistingAzureResourceAnnotation(ApplicationModel.ParameterResource nameParameter, ApplicationModel.ParameterResource? resourceGroupParameter = null) { }
+
+        public ApplicationModel.ParameterResource NameParameter { get { throw null; } }
+
+        public ApplicationModel.ParameterResource? ResourceGroupParameter { get { throw null; } }
+    }
+
+    public static partial class ExistingAzureResourceExtensions
```

Comment:
I think we need an `AsExisting(..)`

Reply:
Most of the APIs we have provide some sort of verb for distinguishing between local dev-scenarios and publish mode-scenarios. It feels a little weird to break from that with this API.

Also, I dunno how much value we provide with an extension method over two lines of code:

```csharp
var builder = DistributedApplication.CreateBuilder();

var resourceName = builder.AddParameter("existingResourceName");
var resourceGroupName = builder.AddParmaeter("existingResourceGroupName");

var sb = builder.AddServiceBus("sb")
  .RunAsExisting(resourceName, resourceGroupName)
  .PublishAsExisting(resourceName, resourceGroupName);
```

Reply:
The scenario might look something like: I am using a key vault in development and one when deployed. They are different but that difference can be parameterized. With your changes today, that looks like duplication.

I think this is a very common scenario, which is why I am asking about it.

Reply:
Yeah agreed on establishing something like `.AsExisting` to cover all contexts. Could apply equally to things like `.AsContainer` for Node apps, etc.